pkg/provider/branch/provider.go

```diff
@@ -158,14 +160,18 @@ func (b *branchENIProvider) InitResource(instance ec2.EC2Instance) error {
 	if err := trunkENI.InitTrunk(instance, podList); err != nil {
 		// If it's an AWS Error, get the exit code without the error message to avoid
 		// broadcasting multiple different messaged events
-		if awsErr, ok := err.(awserr.Error); ok {
+
+		var apiErr smithy.APIError
+
+		if errors.As(err, &apiErr) {
+
 			node, errGetNode := b.apiWrapper.K8sAPI.GetNode(instance.Name())
 			if errGetNode != nil {
 				return fmt.Errorf("failed to get node for event advertisment: %v: %v", errGetNode, err)
 			}
-			var eventMessage = fmt.Sprintf("Failed to create trunk interface: "+
-				"Error Code: %s", awsErr.Code())
-			if awsErr.Code() == "UnauthorizedOperation" {
+			eventMessage := fmt.Sprintf("Failed to create trunk interface: "+
```

Comment:
we should send event for iam error. otherwise we should atleast log the error message and code here.